src/transformers/modeling_blenderbot.py

```diff
@@ -0,0 +1,228 @@
+            Indices of input sequence tokens in the vocabulary.
+            Indices can be obtained using :class:`transformers.BlenderbotTokenizer`.
+            See :func:`transformers.PreTrainedTokenizer.encode` and
+            :func:`transformers.PreTrainedTokenizer.encode_plus` for details.
+            `What are input IDs? <../glossary.html#input-ids>`__
+        encoder_outputs (:obj:`tuple(tuple(torch.FloatTensor)`, `optional`, defaults to :obj:`None`):
+            Tuple consists of (`last_hidden_state`, `optional`: `hidden_states`, `optional`: `attentions`)
+            `last_hidden_state` of shape :obj:`(batch_size, sequence_length, hidden_size)`, `optional`, defaults to :obj:`None`) 
+       attention_mask (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, sequence_length)`, `optional`, defaults to :obj:`None`):
+            Mask to avoid performing attention on padding token indices.
+            Mask values selected in ``[0, 1]``:
+            ``1`` for tokens that are NOT MASKED, ``0`` for MASKED tokens.
+            `What are attention masks? <../glossary.html#attention-mask>`__
+        decoder_input_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, target_sequence_length)`, `optional`, defaults to :obj:`None`):
+        decoder_attention_mask (:obj:`torch.BoolTensor` of shape :obj:`(batch_size, tgt_seq_len)`, `optional`, defaults to :obj:`None`):
+        labels: (:obj:`torch.LongTensor` of shape :obj:`(batch_size, sequence_length)`, `optional`, defaults to :obj:`None`):
+"""
+
+
+class BlenderbotForConditionalGeneration(PretrainedBartModel):
+    config_class = BlenderbotConfig
+    base_model_prefix = "."
+
+    def __init__(self, config: BlenderbotConfig):
+        super().__init__(config)
+        # self.config = config
+        self.shared = nn.Embedding(config.vocab_size, config.d_model, config.pad_token_id)
+        self.encoder = BartEncoder(config, self.shared)
+        self.decoder = BartDecoder(config, self.shared)
+        self.bos_token_id = config.bos_token_id
+        self.init_weights()
+        
+    def output(self, tensor):
+        """
+        Compute output logits.
+        """
+        # project back to vocabulary
+        output = F.linear(tensor, self.shared.weight)
+        # compatibility with fairseq: fairseq sometimes reuses BOS tokens and
+        # we need to force their probability of generation to be 0.
+        output[:, :, self.bos_token_id] = -65504 if output.dtype is torch.float16 else - 1e20
+        return output
+
+    @add_start_docstrings_to_callable(BLENDERBOT_INPUTS_DOCSTRING)
+    def forward(
+        self,
+        input_ids,
+        encoder_outputs=None,
+        decoder_input_ids=None,
+        attention_mask=None,
+        decoder_attention_mask=None,
+        labels=None,
+        decoder_cached_states=None,
+        use_cache=True,
+        output_attentions=None,
+        output_hidden_states=None,
+        return_tuple=None,
```

Comment:
Think this has changed to `return_dict` now (cc @sgugger )

Reply:
Indeed, let me know if you need help for the conversion to `return_dict`, @mariamabarham